cpp/src/io/orc/writer_impl.cu

```diff
@@ -1882,7 +1882,7 @@ encoder_decimal_info decimal_chunk_sizes(orc_table_view& orc_table,
       auto& current_sizes =
         elem_sizes.insert({orc_col.index(), rmm::device_uvector<uint32_t>(orc_col.size(), stream)})
           .first->second;
-      thrust::tabulate(rmm::exec_policy(stream),
+      thrust::tabulate(rmm::exec_policy_nosync(stream),
```

Comment:
When do we know to add `_nosync`?

Reply:
You can find some information here: https://github.com/rapidsai/cudf/issues/12086
With nosync, thrust::tabulate will not sync at the end of execution. This is fine in this case, as the results are on the GPU, and available for the next kernel in the same stream.
We can use this in many, if not all, places in cuIO.